### PR TITLE
Merge `Cms\KirbyTag/s` into `Text\KirbyTag/s`

### PR DESCRIPTION
**Pull request by distantnative**
## Status: ready for review 👀

- Simplifies our code base further

## Breaking changes
- KirbyTag class cannot be set anymore via `static::$tagClass` - I don't think this would affect any users as the KirbyTag classes cannot be replaced anyways (and thus extending the KirbyTags class yourself really doesn't seem like a thing). But open to add it back if you think it's necessary

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] CI checks pass

## When merging

- [ ] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [ ] Add changes to release notes draft in Notion

### CODE REVIEW

**Review by bastianallgeier — DISMISSED** [review]

File: src/Text/KirbyTags.php [hunk 1]
```diff
@@ -20,23 +21,25 @@
  */
 class KirbyTags
 {
-    protected static $tagClass = 'Kirby\Text\KirbyTag';
-
-    public static function parse(string $text = null, array $data = [], array $options = []): string
+    public static function parse(string $text = null, array $data = [], array $options = [], ?App $app = null): string
     {
+        if ($app !== null) {
```
Comment (bastianallgeier) [inline src/Text/KirbyTags.php]: I think we should handle the hooks in the App class in the kirbytags method instead. We already do that in the kirbytext method. I'm not sure why we went this way in the extended class in the first place.

**Comment by distantnative** [comment]
@bastianallgeier now thinking again about it, if we remove `static::$tagClass` as mentioned in the breaking changes, the class cannot really be extended anymore. I think this could be ok, but if we go that way, I would actually mark the class as final.

File: src/Text/KirbyTags.php (hunk 1 above)
Comment (distantnative) [reply src/Text/KirbyTags.php]: moved them

**Review by bastianallgeier — APPROVED** [review]
Perfect!

**Pull request merged**